### DISCUSSION TITLE
Type annotations

### DISCUSSION THREAD

**Discussion by unphased** [mikavilpas/yazi.nvim#395] [Q&A]
I see the `---@type LazySpec` style Lua type annotations in the example config code presented in the README. 

Since my lua-ls lsp in neovim has no idea where these are defined I am curious to learn more about how I might go about setting these up. For example it's not clear to me why or how I could expect such types to get defined in my plugins.lua script, given the script defines what plugins to load and there's a catch-22 here for it to have any access to the types FROM the plugins it's specifying.

**Comment by mikavilpas** [comment]
It's a type in lazy.nvim, maybe it should be loaded automatically?

TBH I don't fully understand how it gets loaded, maybe it's something done in https://github.com/folke/lazydev.nvim

**Reply by unphased** [reply]
That would be fine, but i guess my example I should have used was `---@type YaziConfig`. Where would I even get it? i.e., how would lazy at the loading plugins stage have access to any yazi related types.

**Reply by mikavilpas** [reply]
I'm not exactly sure what you are asking, but if you have yazi.nvim on your runtimepath then something like lazydev can instruct the lua language server to load the types included in it.

If you have not downloaded yazi.nvim yet, then I think the types will not be available. In that case, just download it and maybe restart neovim, and then you should have the types too.
